vividus-soft-assert/src/main/java/org/vividus/softassert/SoftAssertFailableSupplier.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright 2019-2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.vividus.softassert;
+
+@FunctionalInterface
+public interface SoftAssertFailableSupplier<R, E extends Exception>
```

Comment:
```suggestion
public interface FailableSupplier<R, E extends Exception>
```
name can be generic here